File: src/main/kotlin/com/example/toylabs/application/util/ApiResponseUtil.java
package com.example.toylabs.application.util;

import com.example.toylabs.application.dto.common.ApiResponse;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class ApiResponseUtil {

    private ApiResponseUtil() {
    }

    // 성공 응답 (데이터만 포함)
    public static ResponseEntity<ApiResponse> ok(Object data) {
        return ResponseEntity.ok(new ApiResponse(data));
    }

    // 성공 응답 (데이터 없음)
    public static ResponseEntity<ApiResponse> ok() {
        return ResponseEntity.ok(new ApiResponse(null));
    }

    // 실패 응답 (에러 메시지 포함)
    public static ResponseEntity<ApiResponse> fail(String message, HttpStatus status) {
        return ResponseEntity.status(status).body(new ApiResponse(message, status));
    }

}

File: src/main/kotlin/com/example/toylabs/application/util/ApiResponseUtil.kt
package com.example.toylabs.application.util

import com.example.toylabs.application.dto.common.ApiResponse
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

fun <T> T.toSuccessResponse(): ResponseEntity<ApiResponse> { // 기존 success()
    return ResponseEntity.ok(ApiResponse.success(this))
}

fun defaultSuccessResponse(): ResponseEntity<ApiResponse> { // 기존 defaultSuccess()
    return ResponseEntity.ok(ApiResponse.default())
}

fun errorResponse(code: HttpStatus, message: String): ResponseEntity<ApiResponse> { // 기존 error()
    return ResponseEntity.status(code).body(ApiResponse.error(code, message))
}